int32 num
int32 len
int32 width
int32 height
normal_descriptor_node/ndesc[] descriptors